frontend/pages/hosts/ManageHostsPage/ManageHostsPage.tsx

```diff
@@ -146,32 +145,24 @@ const ManageHostsPage = ({
     isOnlyObserver,
     isPremiumTier,
     isFreeTier,
-    setAvailableTeams,
     setCurrentTeam,
-    setCurrentUser,
   } = useContext(AppContext);
   const { renderFlash } = useContext(NotificationContext);
 
-  useQuery(["me"], () => usersAPI.me(), {
-    onSuccess: ({ user, available_teams }: IGetMeResponse) => {
-      setCurrentUser(user);
-      setAvailableTeams(available_teams);
-      if (queryParams.team_id) {
-        const teamIdParam = parseInt(queryParams.team_id, 10);
-        if (
-          isNaN(teamIdParam) ||
-          (teamIdParam &&
-            available_teams &&
-            !available_teams.find((t) => t.id === teamIdParam))
-        ) {
-          router.replace({
-            pathname: location.pathname,
-            query: omit(queryParams, "team_id"),
-          });
-        }
-      }
-    },
-  });
+  if (queryParams.team_id) {
+    const teamIdParam = parseInt(queryParams.team_id, 10);
+    if (
+      isNaN(teamIdParam) ||
+      (teamIdParam &&
+        availableTeams &&
+        !availableTeams.find((team) => team.id === teamIdParam))
+    ) {
+      router.replace({
+        pathname: location.pathname,
+        query: omit(queryParams, "team_id"),
+      });
+    }
+  }
```

Comment:
We no longer need to get /me or /config on load because we can assume it is fresh on every route.